src/plugins/dataflow/nodes/utilities/shared-program-data-utilities.ts

```diff
@@ -10,10 +10,33 @@ export function convertBaseNodeToSharedNode(node: IBaseNode): ISharedProgramNode
       (nodeStateData as any)[key] = (node.model as any)[key];
     }
   });
+
+  // Many node types have particular display methods we need to access to pass a good value to shared program nodes
+  let miniNodeDisplayValue = "";
+  if (node.model.type === "Sensor" && typeof node.getDisplayValue !== 'undefined') {
+    miniNodeDisplayValue = node.getDisplayValue();
+  } else if (node.model.type === "Live Output" && typeof node.getDisplayMessage !== 'undefined') {
+    miniNodeDisplayValue = node.getDisplayMessage();
+  } else if (node.model.type === "Demo Output" && typeof node.getNodeValueDisplayMessage !== 'undefined') {
+    miniNodeDisplayValue = node.getNodeValueDisplayMessage();
+  } else if (node.model.type === "Timer" && typeof node.getSentence !== 'undefined') {
+    miniNodeDisplayValue = node.getSentence();
+  } else {
+    // if NaN or undefined, display empty string
+    if (typeof node.model.nodeValue === 'undefined') {
+      miniNodeDisplayValue = "";
+    } else if (node.model.nodeValue && isNaN(node.model.nodeValue)) {
+      miniNodeDisplayValue = "";
+    } else if (isFinite(node.model.nodeValue)) {
+      const modelVal = node.model.nodeValue;
+      miniNodeDisplayValue = Number.isInteger(modelVal) ? modelVal.toString() : modelVal.toFixed(2);
+    }
+  }
+
```

Comment:
@scytacki - this is certainly inelegant, but works to get the correct display value to the shared program data.  An alternative I considered would be to define a common interface on node classes that needed it, and then add that as an optional property on IBaseNode, and just check for that (or even make it required and not have to check).  Each node type would implement it something like 

```ts
class DemoOutputNode {
  // ...
  getDisplayFunction() {
    return this.getNodeValueDisplayMessage;
  }
}

class TimerNode  {
  // ...
  getDisplayFunction() {
    return this.getSentence;
  }
}
```

Reply:
and then, above, I'd just have to do
```ts
if (typeof node.getDisplayFunction !== 'undefined') miniNodeDisplayValue = node.getDisplayFunction()
```

Reply:
Yeah I think this should be abstracted. I'd recommend adding a method like `getSharedProgramNodeValue()` to BaseNode, and its implementation can be the code in the `else` block above. 

Then each of nodes that need to customize it can override this with their own implementation. This is also where you can handle not including the "closed" in the output block.